managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedger.java

```diff
@@ -292,6 +292,13 @@ ManagedCursor newNonDurableCursor(Position startPosition, String subscriptionNam
      */
     void deleteCursor(String name) throws InterruptedException, ManagedLedgerException;
 
+    /**
+     * Remove a ManagedCursor from this ManagedLedger's waitingCursors.
+     *
+     * @param cursor the ManagedCursor
+     */
+    void removeWaitingCursor(ManagedCursor cursor);
```

Comment:
Is it better to improve the method `deactivateCursor(ManagedCursor cursor)` to remove the cursor from the `waitingCursors`? Looks like we have removed the cursor from the `activeCursors` but does not remove it from the `waitingCursors`, so that we don't need to introduce a new method for ManagedLedger.

Reply:
`deactivateCursor` is widely used not only by closeConsumer, We cannot change the behavior of `deactivateCursor`.

Reply:
@codelipenghui